dash-pipeline/SAI/templates/utils.cpp.j2

```diff
@@ -94,38 +96,46 @@ grpc::StatusCode MutateTableEntry(std::shared_ptr<p4::v1::TableEntry> entry, p4:
     return status.error_code();
 }
 
-bool InsertInTable(std::shared_ptr<p4::v1::TableEntry> entry, sai_object_id_t *objId)
+bool InsertInTable(std::shared_ptr<p4::v1::TableEntry> entry, sai_object_type_t objectType, sai_object_id_t *objId)
 {
     DASH_LOG_ENTER();
 
     auto retCode = MutateTableEntry(entry, p4::v1::Update_Type_INSERT);
-    if (grpc::StatusCode::OK != retCode) {
+
+    if (grpc::StatusCode::OK != retCode)
+    {
         return false;
     }
 
-    std::lock_guard<std::mutex> lock(tableLock);
+    MUTEX;
 
     if (*objId == 0)
     {
-        *objId = NextObjIndex();
+        *objId = NextObjectId(objectType);
     }
+
     tableEntryMap.insert(std::make_pair(*objId, entry));
+
     return true;
 }
 
-sai_object_id_t NextObjIndex()
+sai_object_id_t NextObjectId(sai_object_type_t objectType)
 {
     DASH_LOG_ENTER();
 
+    // TODO make proper oid
+
     return ++nextId;
```

Comment:
If you're going to fix this, shouldn't there be a `nextId` per object type?

Reply:
i left this, since i didn't know whether tests will pass, but updated and fixed

Reply:
>i didn't know whether tests will pass

I looked at your fork, you do not have CI actions enabled. You can enable them and perform all CI testing *prior* to making a PR. This has several advantages - you can try uncertain things out without consequence (and without alerting/distracting watchers of the main repo), you can ensure your PR will pass before you even submit it, etc. I've advised all DASH contributors to do this. It runs in Azure cloud for free, same as DASH CI.

Reply:
thats sounds good, how i can enable them ?

Reply:
Just go to your forked repo, select Settings gear on top ribbon, then click Actions on the left, then you'll see a dialog box where you can enable actions. 
![gitactions](https://github.com/sonic-net/DASH/assets/31145757/18096c1f-26e0-43c3-a8ea-463d84f3b17a)